wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -288,50 +222,32 @@ def run(
                 f"{LOG_PREFIX}Note: no resource args specified. Add a Kubernetes yaml spec or other options in a json file with --resource-args <json>."
             )
         _logger.info(f"Running Kubernetes job with resource args: {resource_args}")
+
         context, api_client = get_kube_context_and_api_client(kubernetes, resource_args)
 
         batch_api = kubernetes.client.BatchV1Api(api_client)
         core_api = kubernetes.client.CoreV1Api(api_client)
 
-        # allow users to specify template or entire spec
-        if resource_args.get("job_spec"):
-            job_dict = load_json_yaml_dict(resource_args["job_spec"])
-        else:
-            # begin constructing job sped
-            job_dict = {"apiVersion": "batch/v1", "kind": "Job"}
+        job: Dict[str, Any] = {"apiVersion": "batch/v1", "kind": "Job"}
+        job.update(resource_args)
```

Comment:
Its kind of weird that we take the job, update it with the raw resource_agrs, then break it down and inject our image, then reassemble the same variable no?

Reply:
It's kind of verbose, but that's how we ensure all the required keys are present. Doing it this way is at least less verbose than repeatedly checking if a key is present and adding it if not (it'd have to be done five times, for `spec`, `template`, `spec`, `containers`, and job `metadata`). I'll see if I can find a simpler way to do this.